src/common/mfu_daos.c

```diff
@@ -4766,4 +4768,176 @@ int daos_cont_deserialize_hdlr(int rank, daos_args_t *da, const char *h5filename
+        if (len > FILENAME_LEN) {
+            MFU_LOG(MFU_LOG_ERR, "destination path exceeds max length "
+                    MFU_ERRF, MFU_ERRP(-MFU_ERR_DCP));
+        }
+    } else {
+        /* destination path is POSIX */
+        rc = 0;
+    }
+
+    pid_t child;
+    int child_status;
+    pid_t child_pid;
+    char *args[4];
+    args[0] = "h5repack";
+
+    if (src_daos && dst_daos) {
+        /* this must be a DAOS->DAOS copy */
+        args[1] = strdup(src_path);
+        args[2] = strdup(dst_path);
+        args[3] = NULL;
+    } else if (src_daos && !dst_daos) {
+        args[1] = "--dst-vol-name=native";
+        args[2] = strdup(src_path);
+        args[3] = strdup(argpaths[1]);
+        args[4] = NULL; 
+    } else if (!src_daos && dst_daos) {
+        args[1] = "--src-vol-name=native";
+        args[2] = strdup(argpaths[0]);
+        args[3] = strdup(dst_path);
+        args[4] = NULL; 
+    } else {
+        MFU_LOG(MFU_LOG_ERR, "Invalid Format " MFU_ERRF,
+                MFU_ERRP(-MFU_ERR_INVAL_ARG));
+        rc = 1;
+        return rc;
+    }
+
+    /* Child process */
+    if ((child = fork()) == 0) {
+
+        /* start h5repack */
+        execvp(args[0], args);
+
+        /* if child process reaches this point then execvp must have failed */
+        MFU_LOG(MFU_LOG_ERR, "execvp on h5repack failed: "
+                MFU_ERRF, MFU_ERRP(-MFU_ERR_DCP));
+        if (src_daos && dst_daos) {
+            mfu_free(&args[1]);
+        } else {
+            mfu_free(&args[3]);
+        }
+        mfu_free(&args[2]);
+        rc = 1;
+        return rc;
+    /* Parent process */
+    } else {
+        if (child == (pid_t)(-1)) {
+            MFU_LOG(MFU_LOG_ERR, "fork failed: "
+                    MFU_ERRF, MFU_ERRP(-MFU_ERR_DCP));
+            if (src_daos && dst_daos) {
+                mfu_free(&args[1]);
+            } else {
+                mfu_free(&args[3]);
+            }
+            mfu_free(&args[2]);
+            rc = 1;
+            return rc;
+        } else {
+            /* parent will wait for child to complete */
+            child_pid = wait(&child_status);
+        }
+    }
+
+    /* if child status is 0, then copy was successful */
+    if (child_status != 0) {
+        MFU_LOG(MFU_LOG_ERR, "errors occurred while copying to/from container: "
+                MFU_ERRF, MFU_ERRP(-MFU_ERR_DCP));
+        if (src_daos && dst_daos) {
+            mfu_free(&args[1]);
+        } else {
+            mfu_free(&args[3]);
+        }
+        mfu_free(&args[2]);
+        rc = 1;
+        return rc;
+    } else {
+        if (dst_daos) {
+            MFU_LOG(MFU_LOG_INFO, "Successfully copied to %s", dst_path);
+        } else {
+            MFU_LOG(MFU_LOG_INFO, "Successfully copied to %s", argpaths[1]);
+        }
+    }
+
+    if (src_daos && dst_daos) {
+        mfu_free(&args[1]);
+    } else {
+        mfu_free(&args[3]);
+    }
+    mfu_free(&args[2]);
```

Comment:
Yeah, I was thinking of how to simplify this too.  Another option is to just always strdup and always free which you can do so long as all pointers are initialize to NULL if not being used:
```
args[1] = strdup("--dst-vol-name=native");
...
mfu_free(&args[1]);
mfu_free(&args[2]);
mfu_free(&args[3]);
mfu_free(&args[4]);
```
That works, because this is valid:
```
args[4] = NULL;
mfu_free(&args[4]);
```

Reply:
Ah, yes. I think that's even simpler!

Reply:
@adammoody @daltonbohning ok, I updated this